### PR TITLE
[DSS] DDP-7291: Add new activity instance status type: ERROR_PRESENT

### PR DESCRIPTION
**Pull request by ssettipalli**
look at https://broadinstitute.atlassian.net/browse/DDP-7291
schema changes 
add new flag / column to enable/disable this new status type for a study.
Add new status type and translations

study-builder support to handle new config parameter to enable/disable study 

DTO/Dao changes and PutFormAnswerRoute changes.

### CODE REVIEW

File: pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/model/activity/types/InstanceStatusType.java [hunk 1]
```diff
@@ -6,5 +6,6 @@
 public enum InstanceStatusType {
     CREATED,
     IN_PROGRESS,
-    COMPLETE
+    COMPLETE,
+    ERROR_PRESENT
```
Comment (ssettipalli) [inline pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/model/activity/types/InstanceStatusType.java:10]: new status type

File: pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/route/PutFormAnswersRoute.java [hunk 2]
```diff
@@ -150,10 +152,27 @@ public Object handle(Request request, Response response) {
...
                     List<ActivityValidationFailure> validationFailures = actValidationService.validate(
                             handle, interpreter, userGuid, operatorGuid, instanceGuid, form.getCreatedAtMillis(),
                             form.getActivityId(), preferredUserLangDto.getId());
                     if (!validationFailures.isEmpty()) {
+                        //check if study has errorPresentStatus enabled
+                        StudyDto studyDto = new JdbiUmbrellaStudyCached(handle).findByStudyGuid(studyGuid);
+                        if (studyDto.isErrorPresentStatusEnabled()) {
+                            instanceStatusDao.updateOrInsertStatus(instanceDto, InstanceStatusType.ERROR_PRESENT,
+                                    Instant.now().toEpochMilli(), operatorUser, participantUser);
+                            return new PutAnswersResponse(WorkflowResponse.unknown());
```
Comment (ssettipalli) [inline pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/route/PutFormAnswersRoute.java:173]: if validations fail and ERROR_PRESENT status is enabled for this study, update activity to this status and return.
angular will goto dashboard.

File: pepper-apis/dss-database/src/main/resources/db-changes/schema/DDP-7291-add-error-present-status-flag.xml [hunk 3]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<databaseChangeLog
+        xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog
+        http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.5.xsd">
+
+    <changeSet author="sampath" id="20220419-add-error-present-status">
```
Comment (ssettipalli) [inline pepper-apis/dss-database/src/main/resources/db-changes/schema/DDP-7291-add-error-present-status-flag.xml:8]: schema change: add new flag to enable/disable new status type

File: pepper-apis/dss-database/src/main/resources/db-changes/seed/DDP-7291-add-error-present-activity-instance-status.xml [hunk 4]
```diff
@@ -0,0 +1,37 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (ssettipalli) [inline pepper-apis/dss-database/src/main/resources/db-changes/seed/DDP-7291-add-error-present-activity-instance-status.xml:1]: added new status translations. en, es, ru for now as per basil support.
need to add more depending on languages supported by fon

**Review by Trimax — APPROVED** [review]
Looks good to me

**Pull request merged**